docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+---
+sidebar_position: 1
+---
+
+# DynamoStore Architecture
+
+When it comes to building an event store on top of a serverless NoSQL database
+it is imperative to apply mechanical sympathy. We've seen implementations of
+DynamoDB event stores using a single-event-per-row strategy. This would seem the
+obvious choice so let's start by explaining why it's not the greatest idea.
+
+## DynamoDB cost model
+
+A great article to understand DynamoDB's pricing model was written by Zach
+Charles
+[here](https://zaccharles.medium.com/calculating-a-dynamodb-items-size-and-consumed-capacity-d1728942eb7c).
+
+The gist of it is that you pay per 4KB for reads, and 1KB for writes, and
+transactional writes will cost you double. As such DynamoStore is designed to
+utilise GetItem and PutItem APIs where possible and only does a table scan and
+transactional write when absolutely necessary.
+
+With these constraints enumerated it should be clear why an event-per-row
+schema is not the greatest idea
+
+1. Reading a stream necessitates a scan in which you pay 0.5RUs at minimum for
```

Comment:
I agree with the point but I dont think you actually pay per row so people won't accept the point - it is charged based on the aggregated size. (In Cosmos I'd argue less, the headers are big and always travel but in DDB they dont)

Reply:
Yeah I'm wrong on this, DDB stores items with the same partition key together and you pay for the data returned from the query